src/codemagic/tools/_app_store_connect/action_groups/builds_action_group.py

```diff
@@ -68,6 +69,44 @@ def get_build(self, build_id: ResourceId, should_print: bool = True) -> Build:
 
         return self._get_resource(build_id, self.api_client.builds, should_print)
 
+    @cli.action(
+        'expire',
+        BuildArgument.BUILD_ID_RESOURCE_ID,
+        action_group=AppStoreConnectActionGroup.BUILDS,
+    )
+    def expire_build(self, build_id: ResourceId, should_print: bool = True) -> Build:
+        """
+        Expire a specific build, an expired build becomes unavailable for testing
+        """
+
+        return self._modify_resource(
+            self.api_client.builds,
+            build_id,
+            should_print,
+            expired=True,
+        )
+
+    @cli.action(
+        'expire-previous-builds',
+        AppArgument.APPLICATION_ID_RESOURCE_ID,
+        BuildArgument.BUILD_ID_RESOURCE_ID,
+        action_group=AppStoreConnectActionGroup.BUILDS,
+    )
+    def expire_previous_builds(
+        self, application_id: ResourceId, build_id: ResourceId, should_print: bool = True,
+    ) -> List[Build]:
+        """
+        Expire all builds except the given build
+        """
+        builds = self.list_builds(application_id=application_id, not_expired=True, should_print=should_print)
+        return [
+            self.expire_build(
+                build_id=asc_build.id,
+            )
+            for asc_build in builds
```

Comment:
Why not just `for build in builds`? The prefix `asc` does not give much additional information.

Reply:
I initially set it because I had the condition `build.id != build_id`, and I misread something with that at some point, so I wanted to separate them a bit, but yes, all-in-all, it probably does not make much sense.